File: backend/app/model/message.py

from datetime import datetime
from enum import Enum
from typing import Optional
from app.core.firestore_repository import FirestoreRepository, collection

class MessageState(str, Enum):
    REPLIED = "replied"
    INTENT_NOT_FOUND = "intent_not_found"
    BELOW_THRESHOLD = "below_threshold"

@collection("message_logs")
class MessageRepository(FirestoreRepository):
    uid: str
    text: str
    created_on: datetime
    tag: str
    probability: float
    response: Optional[str]
    state: MessageState
    threshold: float

File: admin/src/models/message.ts
import { Timestamp } from "firebase/firestore";

export enum MessageState {
	REPLIED = "replied",
	INTENT_NOT_FOUND = "intent_not_found",
	BELOW_THRESHOLD = "below_threshold"
}

interface Message {
	document_id: string;
	uid: string;
	text: string;
	created_on: Timestamp;
	tag: string;
	probability: string;
	response?: string;
	state: MessageState;
	threshold: number;
}

export default Message;